src/breakpoints/index.ts

```diff
@@ -43,17 +47,160 @@ export class Breakpoints extends Panel {
         iconClassName: 'jp-CloseAllIcon',
         onClick: () => {
           this.model.breakpoints = [];
+          this.cellsBreakpoints[this.getCell().id] = [];
+          this.removeAllGutterBreakpoints(this.getCell());
         },
         tooltip: 'Remove All Breakpoints'
       })
     );
+
+    this.noteTracker = options.noteTracker;
+    if (this.noteTracker) {
+      this.noteTracker.activeCellChanged.connect(
+        this.onActiveCellChanged,
+        this
+      );
+    }
   }
 
   private isAllActive = true;
   readonly body: Widget;
   readonly model: Breakpoints.IModel;
-}
+  noteTracker: INotebookTracker;
+  previousCell: CodeCell;
+  previousLineCount: number;
+  cellsBreakpoints: { [id: string]: Breakpoints.IBreakpoint[] } = {};
```

Comment:
Could we change the name of this to `cellBreakpoints` where `cell` is singular, please?